src/snowflake/snowpark/stored_procedure.py

```diff
@@ -390,6 +397,7 @@ def register(
             replace,
             parallel,
             statement_params=statement_params,
+            execute_as=execute_as,
```

Comment:
Should we check the value of `execute_as` in `_do_register_sp` to be either `owner` or `caller`? This way it also makes negative test easier.